src/yunohost/tools.py

```diff
@@ -982,208 +1019,206 @@ def tools_migrations_list(pending=False, done=False):
-                logger.success(m18n.n('migrations_success',
-                                      number=migration.number, name=migration.name))
-
-        else:  # if skip
-            logger.warn(m18n.n('migrations_skip_migration',
-                               number=migration.number,
-                               name=migration.name))
-
-        # update the state to include the latest run migration
-        state["last_run_migration"] = {
-            "number": migration.number,
-            "name": migration.name
-        }
-
-        operation_logger.success()
-
-        # Skip migrations one at a time
-        if skip:
-            break
-
-    # special case where we want to go back from the start
-    if target == 0:
-        state["last_run_migration"] = None
+                if revert:
+                    logger.success(m18n.n('migrations_success_revert', id=migration.id))
+                    migration.state = "pending"
+                    _write_migration_state(migration.id, "pending")
+                else:
+                    logger.success(m18n.n('migrations_success_forward', id=migration.id))
+                    migration.state = "done"
+                    _write_migration_state(migration.id, "done")
 
-    write_to_json(MIGRATIONS_STATE_PATH, state)
+                operation_logger.success()
 
 
 def tools_migrations_state():
     """
     Show current migration state
     """
+    if os.path.exists("/etc/yunohost/migrations_state.json"):
+        _migrate_legacy_migration_json()
+
     if not os.path.exists(MIGRATIONS_STATE_PATH):
-        return {"last_run_migration": None}
+        return {"migrations": {}}
 
-    return read_json(MIGRATIONS_STATE_PATH)
+    return read_yaml(MIGRATIONS_STATE_PATH)
 
 
-def tools_shell(command=None):
-    """
-    Launch an (i)python shell in the YunoHost context.
+def _migrate_legacy_migration_json():
 
-    This is entirely aim for development.
-    """
+    from moulinette.utils.filesystem import read_json
 
-    from yunohost.utils.ldap import _get_ldap_interface
-    ldap = _get_ldap_interface()
+    logger.debug("Migrating legacy migration state json to yaml...")
 
-    if command:
-        exec(command)
-        return
+    # We fetch the old state containing the last run migration
+    old_state = read_json("/etc/yunohost/migrations_state.json")["last_run_migration"]
+    last_run_migration_id = str(old_state["number"]) + "_" + old_state["name"]
 
-    logger.warn("The \033[1;34mldap\033[0m interface is available in this context")
+    # Extract the list of migration ids
     try:
-        from IPython import embed
-        embed()
+        from . import data_migrations
     except ImportError:
-        logger.warn("You don't have IPython installed, consider installing it as it is way better than the standard shell.")
-        logger.warn("Falling back on the standard shell.")
+        # Well ugh what are we supposed to do if we can't do this >.>...
+        pass
```

Comment:
Probably should explode here!

Reply:
Personally i think we should remove the try except here

Reply:
You mean we kind safely assume that we can perform the import because that would be really strange otherwise ? I kind of agree, e.g. we don't do `try ... except` for the usual imports at the top of the file either ..